src/marlin_stubs/sdcard/M20-M30_M32-M34.cpp

```diff
@@ -1,22 +1,77 @@
 #include <dirent.h>
 
 #include "../../lib/Marlin/Marlin/src/gcode/gcode.h"
+#include "libs/hex_print_routines.h"
 #include "marlin_server.hpp"
 #include "media.h"
 #include "marlin_vars.hpp"
 
-// M20 - List SD card
-void GcodeSuite::M20() {
-    SERIAL_ECHOLNPGM(MSG_BEGIN_FILE_LIST);
+static void print_file_size(const char *const filename) {
+    struct stat stat_buf;
+    int rc = stat(filename, &stat_buf);
+    if (rc == 0) {
+        SERIAL_ECHO(stat_buf.st_size);
+    }
+}
+
+static void print_file_info(const char *const dirname, const char *const long_dirname, bool timestamps, bool long_filenames) {
```

Comment:
Just realized that dirname is actually filename.

Reply:
It's not just filename, it does hold the prefix "/usb/" as well, wo maybe I should name it path instead

Reply:
It's now named path and long_path, which is appropriate.